sklearn/tests/test_common.py

```diff
@@ -217,6 +217,9 @@ def test_import_all_consistency():
     for modname in submods + ["sklearn"]:
         if ".tests." in modname:
             continue
+        # Avoid test suite depending on setuptools
```

Comment:
My understanding is that it will indeed try to import all the submodules, but that `ImportError` is going to be ignored. By default (`onerror=None`), only `ImportError` is ignored. When specifying `onerror`, any kind of `Exception` is ignored:
https://github.com/python/cpython/blob/1f885df2a580360c5de69cc41191f3c6bfaaeb35/Lib/pkgutil.py#L76-L86